src/west/_bootstrap/main.py

```diff
@@ -196,44 +230,118 @@ def init_bootstrap(directory, args):
     # Clone the west source code and the manifest into west/. Git will create
     # the west/ directory if it does not exist.
 
-    clone('west repository', args.west_url, args.west_rev,
+    clone('west repository', west_url, west_rev,
           os.path.join(directory, WEST_DIR, WEST))
 
-    clone('manifest repository', args.manifest_url, args.manifest_rev,
+    clone('manifest repository', manifest_url, manifest_rev,
           os.path.join(directory, WEST_DIR, MANIFEST))
 
     # Create an initial configuration file
 
-    config = configparser.ConfigParser()
+    config_path = os.path.join(directory, WEST_DIR, 'config')
+    update_conf(config_path, west_url, west_rev, manifest_url, manifest_rev)
+    print('=== Initial configuration written to {} ==='.format(config_path))
 
-    config['west'] = {
-        'remote': args.west_url,
-        'revision': args.west_rev
-    }
+    # Create a dotfile to mark the installation. Hide it on Windows.
 
-    config['manifest'] = {
-        'remote': args.manifest_url,
-        'revision': args.manifest_rev
-    }
+    with open(os.path.join(directory, WEST_DIR, WEST_MARKER), 'w') as f:
+        hide_file(f.name)
 
-    config_path = os.path.join(directory, WEST_DIR, 'config')
+    print('=== West initialized ===')
+
+
+def reinit(config_path, args):
+    '''
+    Reinitialize an existing installation.
+
+    Currently, this is just a shorthand for updating the west/config
+    configuration file.
+    '''
+    if args.base_url:
+        if args.west_url or args.manifest_url:
+            sys.exit('fatal error: -b is incompatible with -m and -w')
+        west_url = args.base_url.rstrip('/') + '/west'
+        manifest_url = args.base_url.rstrip('/') + '/manifest'
+    else:
+        west_url = args.west_url
+        manifest_url = args.manifest_url
+
+    if not (west_url or args.west_rev or manifest_url or args.manifest_rev):
+        sys.exit('West already initialized. Please pass any settings you '
```

Comment:
Could running it without any changes possibly be useful?

For example, if the user has accidentally deleted the west directory and tries to run `west init` again to fix it. It seems like it would be more helpful to go ahead and do the reinit anyway than stop here even if nothing "seems" different.

What would the downside of doing that be?

Reply:
I'm also running into a situation where I have pre-0.3.0 installations around that don't have a west/config. Being able to re-run west init to create one would be a nice feature.

If there's no downside to be avoided that I'm missing, I'm not sure this exit is helpful.

Reply:
Guess `west init` would still re-initialize everything if west/ has been deleted, since there's no west installation anymore.

Reply:
The manifest repository still wouldn't get re-cloned without the check though. What it'd do would be to write `=== Updated configuration written to west/config ===`, but actually do nothing (because all the parameters to `update_conf()` are `None`).

I wouldn't be terribly against that to be honest, but I think the message it prints now is nicer.

At some point, I think the answer to "it hurts when I do this" is "don't do it". Randomly deleting installation files might fall there I think. :)

Reply:
One improvement to the bootstrapper might be to have it do the two-step `git init` + `git fetch` thing that the project commands do though, for the manifest and west repositories. That gives you more flexibility in how you resume failed operations and the like.

Think that falls outside the scope of this PR though, and I'm not sure exactly how we'd set it up.